File: Java_Joc_Windows_Android/AndroidStudioProject/core/src/local/profile/PlayerProfileDataLocalPlayerGems.java
package local.profile;

import base.launcher.ILocalDataHandler;
import base.profile.PlayerProfileData;
import local.app.AppLocal;

//The player profile handles persistent game information.
public class PlayerProfileDataLocalPlayerGems extends PlayerProfileData {

	public static final String CURRENCY_PLAYER_GEMS = "CURRENCY_PLAYER_GEMS";

    //asd_002;// dispatch a animated currency value, it is separate: CURRENCY_PLAYER_GEMS_ANIMATED.

    public PlayerProfileDataLocalPlayerGems(AppLocal app, ILocalDataHandler localDataHandler) {
        super(app, localDataHandler);

	    //void
    }

    public int getPlayerGems() {
	    return localDataHandler.getInt(CURRENCY_PLAYER_GEMS, 0);
    }

    public void putPlayerGems(int gems) {
	    localDataHandler.putInt(CURRENCY_PLAYER_GEMS, gems);

	    dispatchEvent_profileDataChanged(CURRENCY_PLAYER_GEMS);
    }

    public void addPlayerGems(int gemsAddValue) {
	    putPlayerGems(getPlayerGems() + gemsAddValue);
    }

}

File: CPP_Joc_Windows_Android/SH_Client_Win_Cpp_Cmake/App/src/base/profiledata/PlayerProfileDataLocalPlayerGems.cpp
#include "PlayerProfileDataLocalPlayerGems.h"

using namespace base;

std::string PlayerProfileDataLocalPlayerGems::CURRENCY_PLAYER_GEMS = "CURRENCY_PLAYER_GEMS";

PlayerProfileDataLocalPlayerGems::~PlayerProfileDataLocalPlayerGems() {
    //void
}

PlayerProfileDataLocalPlayerGems::PlayerProfileDataLocalPlayerGems(ILogHandler* logger, base::IProfileStorage* localDataHandler)
    : super(logger, localDataHandler)
{
    //void
}

int PlayerProfileDataLocalPlayerGems::getPlayerGems() {
    return localDataHandler->getInt(CURRENCY_PLAYER_GEMS, 0);
}

void PlayerProfileDataLocalPlayerGems::putPlayerGems(int gems) {
    localDataHandler->putInt(CURRENCY_PLAYER_GEMS, gems);

    dispatchEvent_profileDataChanged(CURRENCY_PLAYER_GEMS);
}

void PlayerProfileDataLocalPlayerGems::addPlayerGems(int gemsAddValue) {
    putPlayerGems(getPlayerGems() + gemsAddValue);
}
